### PR TITLE
Update CREDITS on a best-effort basis.

### PR DESCRIPTION
**Pull request by Zimmi48**
This file hasn't been updated in quite a while.
It should either be maintained or removed.
I've done some updating on a best-effort basis.
@forestjulien @tchajed @Matafou @ejgallego @SkySkimmer @JasonGross @matejkosik @gmalecha @cmangin @ppedrot @aspiwack @psteckler @amintimany @zeldovich: please check that your institutions are correct.

Note: a mention should be added for SSReflect but I don't know exactly what it should read.
Note: this is not for the beta.

### CODE REVIEW

File: CREDITS [hunk 1]
```diff
@@ -94,32 +97,41 @@ of the Coq Proof assistant during the indicated time:
...
+  Pierre Courtieu (CNAM, 2013-now)
   David Delahaye (INRIA, 1997-2002)
   Maxime Dénès (INRIA, 2013-now)
-  Daniel de Rauglaudre (INRIA, 1996-1998)
+  Daniel de Rauglaudre (INRIA, 1996-1998, 2012, 2016)
   Olivier Desmettre (INRIA, 2001-2003)
   Gilles Dowek (INRIA, 1991-1994)
   Amy Felty (INRIA, 1993)
   Jean-Christophe Filliâtre (ENS Lyon, 1994-1997, LRI, 1997-2008)
+  Emilio Jesús Gallego Arias (MINES ParisTech 2015-now)
```
Comment (ejgallego) [inline CREDITS]: 2014-now

File: CREDITS [hunk 2]
```diff
@@ -7,16 +7,19 @@ The "Coq proof assistant" was jointly developed by
...
   config dev ide interp intf kernel lib library parsing pretyping proofs
   scripts states tactics test-suite theories tools toplevel
 
 are distributed under the terms of the GNU Lesser General Public License
-Version 2.1 (see file LICENSE). These files are COPYRIGHT 1999-2010,
-The Coq development team, CNRS, INRIA and Université Paris Sud.
+Version 2.1 (see file LICENSE). These files are COPYRIGHT 1999-2017,
+The Coq development team, INRIA, CNRS, LIX, LRI, PPS.
 
 Files from the directory doc are distributed as indicated in file doc/LICENCE.
```
Comment (psteckler) [inline CREDITS]: In fact, the file is `doc/LICENSE`.

Comment (psteckler) [reply CREDITS]: I think that file needs updating. The most recent copyright there is 2006.

File: CREDITS (hunk 1 above)
Comment (Zimmi48) [reply CREDITS]: Note that we are marking the dates of the contributions to Coq. So is it 2014-now?

Comment (ejgallego) [reply CREDITS]: Indeed it was early 2015, sorry for the noise.

File: CREDITS [hunk 3]
```diff
@@ -38,7 +41,7 @@ plugins/fourier
   developed by Loïc Pottier (INRIA-Lemme, 2001)
 plugins/funind
   developed by Pierre Courtieu (INRIA-Lemme, 2003-2004, CNAM, 2004-2008),
-  Julien Forest (INRIA-Everest, 2006, CNAM, 2007-2008)
+  Julien Forest (INRIA-Everest, 2006, CNAM, 2007-2008, ENSIIE, 2013-now)
```
Comment (forestjulien) [inline CREDITS]: ENSIIE: 2008-now for me

**Comment by aspiwack** [comment]
Yep, mine's good. Thanks for that.

File: CREDITS [hunk 4]
```diff
@@ -94,32 +97,41 @@ of the Coq Proof assistant during the indicated time:
...
-  Pierre Boutillier (INRIA-PPS, 2010-now)
+  Pierre Boutillier (INRIA-PPS, 2010-2015)
   Xavier Clerc (INRIA, 2012-2014)
+  Tej Chajed (MIT, 2016-now)
   Jacek Chrzaszcz (LRI, 1998-2003)
   Thierry Coquand (INRIA, 1985-1989)
   Pierre Corbineau (LRI, 2003-2005, Nijmegen, 2005-2008, Grenoble 1, 2008-2011)
   Cristina Cornes (INRIA, 1993-1996)
   Yann Coscoy (INRIA Sophia-Antipolis, 1995-1996)
+  Pierre Courtieu (CNAM, 2013-now)
```
Comment (forestjulien) [inline CREDITS]: the dates are not correct but not sure of the good ones

Comment (Zimmi48) [reply CREDITS]: This file reports the dates of the code contributions to Coq. I browsed the commit history to set them (which doesn't mean that there can't be any mistakes).

Comment (Matafou) [inline CREDITS]: Be careful that some people changed their id at some point:
For me I changed from courtieu to Courtieu at some point

git log --pretty=format:"%an %cd" | grep "courtieu\\|Courtieu"

gives commits each years from 2006 up to now. Same for @forestjulien.

I guess 2013 is the year we switched to git. We should have detect this inconsistencies during the migration.

P.

Comment (SkySkimmer) [reply CREDITS]: This information is already available in a machine readable format, although I don't know the precise invocation to use it
https://github.com/coq/coq/blob/master/.mailmap#L28

Comment (Zimmi48) [reply CREDITS]: Yeah, me neither...
Anyways, these dates issues are now fixed.

**Comment by ppedrot** [comment]
As far as I'm concerned, that's OK.

File: CREDITS [hunk 5]
```diff
@@ -94,32 +97,41 @@ of the Coq Proof assistant during the indicated time:
...
+  Emilio Jesús Gallego Arias (MINES ParisTech 2015-now)
+  Gaetan Gilbert (INRIA-CoqHoTT 2016-now)
   Eduardo Giménez (ENS Lyon, 1993-1996, INRIA, 1997-1998)
   Stéphane Glondu (INRIA-PPS, 2007-2013)
   Benjamin Grégoire (INRIA, 2003-2011)
+  Jason Gross (MIT 2013-now)
   Hugo Herbelin (INRIA, 1996-now)
   Sébastien Hinderer (INRIA, 2014)
   Gérard Huet (INRIA, 1985-1997)
+  Matej Košík (INRIA, 2015-now)
```
Comment (matejkosik) [inline CREDITS]: The above line can be replaced with:
```
Matej Košík (INRIA, 2015-2017)
```
I think.

Comment (Zimmi48) [reply CREDITS]: Thanks. Updated.

**Comment by silene** [comment]
> This information is already available in a machine readable format, although I don't know the precise invocation to use it.

Simply set the `log.mailmap` option in your git global configuration. No idea why it is not on by default.

**Comment by maximedenes** [comment]
@Zimmi48 Is this ready for merge?

**Comment by Zimmi48** [comment]
I didn't get feedback yet from @tchajed @JasonGross @gmalecha @cmangin @amintimany @zeldovich, so I would give it a few more days. Of course, if we merge with errors it can also be fixed later.

**Comment by amintimany** [comment]
@Zimmi48: It looks good to me. Thanks :-)

**Comment by maximedenes** [comment]
@Zimmi48 whenever you feel this is ready for merge, please remove the `needs:progress` flag.

**Comment by Zimmi48** [comment]
We are not going to wait forever until people give feedback so let's merge this and if people want to fix stuff afterwards, they can just open a new PR.

**Comment by Zimmi48** [comment]
> Note: a mention should be added for SSReflect but I don't know exactly what it should read.

There is still the issue of SSReflect not being mentioned in [`CREDITS`](https://github.com/coq/coq/blob/master/CREDITS). @gares Could you open a PR fixing this?

**Pull request merged**